server2/routes/signupRouter.js

```diff
@@ -15,30 +17,39 @@ router.route('/')
     })
     .post((req, res, next) => {
         const newUser = req.body;
-        //User.sync()
+        // User.sync()
         Promise.all([isDuplicateEmail(newUser), isDuplicateUser(newUser)])
             .then((newUser) => {
                 const saltRounds = 10;
-                const salt = bcrypt.genSaltSync(saltRounds)
-                const hash = bcrypt.hashSync(req.body.password, salt);
-                
-                const test_user = User.create({
+                const salt = bcrypt.genSaltSync(saltRounds);
+                const hash = bcrypt.hashSync(req.body.password, saltRounds);
+
+                const new_user = {
                     id: idGen(),
-                    username: req.body.username,
+                    name: req.body.username,
```

Comment:
can empty string be a valid user name?

Reply:
this can be checked in frontend. wont be able to click submit if it is empty. shld we oso write this in backend?